packages/salesforcedx-sobjects-faux-generator/src/generator/fauxClassGenerator.ts

```diff
@@ -221,6 +239,82 @@ export class FauxClassGenerator {
     return this.successExit();
   }
 
+  public async generateMin(
+    projectPath: string,
+    source: SObjectRefreshSource
+  ): Promise<SObjectRefreshResult> {
+    this.result = {
+      data: { category: SObjectCategory.STANDARD, source, cancelled: false }
+    };
+    const sobjectsFolderPath = path.join(
+      projectPath,
+      SFDX_DIR,
+      TOOLS_DIR,
+      SOBJECTS_DIR
+    );
+    const standardSObjectsFolderPath = path.join(
+      sobjectsFolderPath,
+      STANDARDOBJECTS_DIR
+    );
+
+    if (
+      !fs.existsSync(projectPath) ||
+      !fs.existsSync(path.join(projectPath, SFDX_PROJECT_FILE))
+    ) {
+      return this.errorExit(
+        nls.localize('no_generate_if_not_in_project', sobjectsFolderPath)
+      );
+    }
+    this.cleanupSObjectFolders(sobjectsFolderPath, SObjectCategory.STANDARD);
+
+    const connection = await Connection.create({
+      authInfo: await AuthInfo.create({
+        username: await ConfigUtil.getUsername(projectPath)
+      })
+    });
+
+    const describe = new SObjectDescribe(connection);
+    const standardSObjects: SObject[] = [];
+    let fetchedSObjects: SObject[] = [];
+    let j = 0;
+    while (j < startupMinSObjects.length) {
+      try {
+        if (
+          this.cancellationToken &&
+          this.cancellationToken.isCancellationRequested
+        ) {
+          return this.cancelExit();
+        }
+        fetchedSObjects = fetchedSObjects.concat(
+          await describe.describeSObjectBatch(startupMinSObjects, j)
```

Comment:
I'm not quite getting how batching works here. In `describeSObjectBatch`, it calls `runRequest` which makes an xhr request, and this block is running `startupMinSObjects.length` times.
https://github.com/forcedotcom/salesforcedx-vscode/blob/1cfa48b0b45da0c47aaa0e466115042ce0f1026c/packages/salesforcedx-sobjects-faux-generator/src/describe/sObjectDescribe.ts#L284

Reply:
This hasn't changed from the previous implementation. The describe calls are in essence paginated with `j` tracking how many objects have been fetched. [This line right below](https://github.com/forcedotcom/salesforcedx-vscode/pull/2467/files#diff-b663fb174d801c41abb1ad0fa66fedc3R291) sets `j` to how many were fetched in the current "page"

Reply:
Got it, so `j` is always equal to the current count of `fetchedSObjects` here. I think from an API perspective this should all be part of `describeSObjectBatch`

Reply:
Since it hasn't changed from original, fine keeping as is